src/Shared/Guard.cs

```diff
@@ -55,161 +55,160 @@ internal sealed class NotNullAttribute : Attribute
 }
 #endif
 
-namespace OpenTelemetry.Internal
+namespace OpenTelemetry.Internal;
```

Comment:
This is erroring out with: `error CS8956: File-scoped namespace must precede all other members in a file.` 

I think we should move `CallerArgumentExpressionAttribute` and `NotNullAttribute` to their own classes.

Reply:
I will sort this out in an hour or two

Reply:
@utpilla I'm having an issue with moving those attributes to their own classes as this is code is kinda unusual, mind guiding me?

Reply:
You could add a new file named `CallerArgumentExpressionAttribute.cs` under `Shared/Shims/` and move this code into that file:

```csharp
#if !NET6_0_OR_GREATER
namespace System.Runtime.CompilerServices
{
    /// <summary>Allows capturing of the expressions passed to a method.</summary>
    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
    internal sealed class CallerArgumentExpressionAttribute : Attribute
    {
        public CallerArgumentExpressionAttribute(string parameterName)
        {
            this.ParameterName = parameterName;
        }

        public string ParameterName { get; }
    }
}
#endif
```

Reply:
Yeah but will Guard.cs pick up those new classes same as when they are in one file?

Reply:
Similarly, you could create a file named `NotNullAttribute.cs` under `Shared/Shims/` and move this code into that file:

```csharp
#if !NET6_0_OR_GREATER && !NETSTANDARD2_1_OR_GREATER
namespace System.Diagnostics.CodeAnalysis
{
    /// <summary>Specifies that an output is not <see langword="null"/> even if
    /// the corresponding type allows it. Specifies that an input argument was
    /// not <see langword="null"/> when the call returns.</summary>
    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter | AttributeTargets.ReturnValue, Inherited = false)]
    internal sealed class NotNullAttribute : Attribute
    {
    }
}
#endif
```

Reply:
> Yeah but will Guard.cs pick up those new classes same as when they are in one file?

`Guard` class can access them because it imports their namespaces:

```csharp
using System.Diagnostics.CodeAnalysis;
using System.Runtime.CompilerServices;
```

Reply:
Doesn't seem to work this way, see the test results after my last push

Reply:
@jarosal, you should probably find all occurrences if Guard.cs file and include new files in the similar way.

See https://github.com/open-telemetry/opentelemetry-dotnet/blob/8f38400a142df32259bd78d3cbeea8fd3c8e867a/src/OpenTelemetry.Exporter.Console/OpenTelemetry.Exporter.Console.csproj#L33

Reply:
@jarosal On second thoughts, could you suppress the warning for using file-scoped namespace in this class?